packages/teleterm/src/ui/DocumentCluster/ClusterResources/Servers/Servers.tsx

```diff
@@ -36,43 +41,77 @@ export default function Container() {
 }
 
 function ServerList(props: State) {
-  const { servers = [], getSshLogins, connect, syncStatus } = props;
+  const {
+    servers = [],
+    getSshLogins,
+    connect,
+    fetchAttempt,
+    agentFilter,
+    pageCount,
+    customSort,
+    prevPage,
+    nextPage,
+    updateQuery,
+    onAgentLabelClick,
+    disabledRows,
+    updateSearch,
+  } = props;
   return (
     <>
-      {syncStatus.status === 'failed' && (
-        <Danger>{syncStatus.statusText}</Danger>
+      {fetchAttempt.status === 'error' && (
+        <Danger>{fetchAttempt.statusText}</Danger>
       )}
-      <Table
-        columns={[
-          {
-            key: 'hostname',
-            headerText: 'Hostname',
-            isSortable: true,
-          },
-          {
-            key: 'addr',
-            headerText: 'Address',
-            isSortable: true,
-            render: renderAddressCell,
-          },
-          {
-            key: 'labelsList',
-            headerText: 'Labels',
-            render: renderLabelCell,
-          },
-          {
-            altKey: 'connect-btn',
-            render: server =>
-              renderConnectCell(
-                () => getSshLogins(server.uri),
-                login => connect(server.uri, login)
-              ),
-          },
-        ]}
-        emptyText="No Nodes Found"
-        data={servers}
-        pagination={{ pageSize: 15, pagerPosition: 'bottom' }}
+      <SearchPanel
+        updateQuery={updateQuery}
+        updateSearch={updateSearch}
+        pageCount={pageCount}
+        filter={agentFilter}
+        showSearchBar={true}
+        disableSearch={disabledRows}
       />
+      <DarkenWhileDisabled className={disabledRows ? 'disabled' : ''}>
+        <Table
+          columns={[
+            {
+              key: 'hostname',
+              headerText: 'Hostname',
+              isSortable: true,
+            },
+            {
+              key: 'addr',
+              headerText: 'Address',
+              isSortable: true,
+              render: renderAddressCell,
+            },
+            {
+              key: 'labelsList',
+              headerText: 'Labels',
+              render: ({ labelsList }) => (
+                <ClickableLabelCell
+                  labels={labelsList}
+                  onClick={onAgentLabelClick}
+                />
+              ),
+            },
+            {
+              altKey: 'connect-btn',
+              render: server =>
+                renderConnectCell(
+                  () => getSshLogins(server.uri),
+                  login => connect(server, login)
+                ),
+            },
+          ]}
+          customSort={customSort}
+          emptyText={
+            fetchAttempt.status === 'success'
+              ? 'No servers found'
+              : 'Searching…'
```

Comment:
If the fetch attempt fails, the tab looks like this:

![failed fetch](https://user-images.githubusercontent.com/27113/201122915-e3b0f59b-b58a-4cf2-8d05-bc9c0dc1bf90.png)

We could change this to `(!fetchAttempt.status || fetchAttempt.status === 'loading') ? 'Searching…' : 'No servers found'` but then it'd say "No servers found" on error too. In that situation it'd be better if it just said "Failed to fetch servers".

And since it means the conditional has to be much bigger, IMHO it'd be better to define it above as a variable, let's say `tableEmptyText`, and pass it as a prop here. FWIW, I like to use the switch statement when operation on types such as `fetchAttempt.status`.

Reply:
I made a switch statement inside `useServers` instead of `useServersideResources` because I think it will give us greater flexibility in the text we return "No _servers_ found" as an example. Also, `useServersideResources` probably shouldn't care how the data is used and `useServers` is more like "ok lets fill a Table specifically with this info and here is the empty text for said table". 

What I'm getting at is: Great idea!! 👍

Reply:
> I made a switch statement inside `useServers` instead of `useServersideResources` because I think it will give us greater flexibility in the text we return "No _servers_ found" as an example. Also, `useServersideResources` probably shouldn't care how the data is used and `useServers` is more like "ok lets fill a Table specifically with this info and here is the empty text for said table".

Good move and the rationale sounds very good too. 👍